File: examples/sharing/deny-access-request.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Deny a pending document access request. Owner/admin only. The optional
// `documentUrl` is included in the requester's notification email.
export async function denyAccessRequest(
  client: JsBaoClient,
  documentId: string,
  requestId: string,
) {
  // #region example
  await client.documents.denyAccessRequest(documentId, requestId, {
    documentUrl: "https://myapp.example/docs/" + documentId,
  });
  // #endregion example
}

File: examples/sharing/deny-access-request.swift
import JsBaoClient

// Deny a pending document access request. Owner/admin only. The optional
// `documentUrl` is included in the requester's notification email.
func denyAccessRequest(
  client: JsBaoClient,
  documentId: String,
  requestId: String
) async throws {
  // #region example
  _ = try await client.documents.denyAccessRequest(
    documentId: documentId,
    requestId: requestId,
    params: ["documentUrl": "https://myapp.example/docs/\(documentId)"]
  )
  // #endregion example
}
